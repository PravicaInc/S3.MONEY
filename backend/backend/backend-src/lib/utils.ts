/**
 * Convert smart contract ticker to package name.
 *
 * @param {string} ticker - smart contract ticker, e.g. $SUI
 * @returns {string} - package name, e.g. sui
 */
export function tickerToPackageName(ticker: string): string {
  return ticker.toLowerCase().trim()
    .slice(1);
}

// eof
