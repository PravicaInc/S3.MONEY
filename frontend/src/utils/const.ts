export const PAGES_URLS = {
  home: '/',
  dashboardOverview: '/dashboard/overview',
  dashboardOperations: '/dashboard/operations',
  signIn: '/sign-in',
  createStableCoin: '/create-stablecoin',
};

export const WALLETS = {
  SuiWallet: 'Sui Wallet',
};
