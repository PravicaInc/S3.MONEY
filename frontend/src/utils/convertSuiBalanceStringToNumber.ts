const convertSuiBalanceStringToNumber = (balance: string): number => parseInt(balance) / 10e8;

export default convertSuiBalanceStringToNumber;
