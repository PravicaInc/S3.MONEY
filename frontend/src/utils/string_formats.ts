export const numberNormalize = (value: string = ''): number => Number(`${value}`.replace(/\D+/g, ''));

export const numberFormat = (value: string): string => value
  ? new Intl.NumberFormat('en-US').format(numberNormalize(value))
  : value;

export const priceFormat = (value: string): string => value
  ? new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    maximumFractionDigits: 2,
    minimumFractionDigits: value.indexOf('.') !== -1 ? 2 : 0,
  })
    .format(numberNormalize(value))
  : value;

export const getShortAccountAddress = (accountAddress: string, partLength: number = 4) => (
  `${accountAddress.substring(0, partLength)}...${accountAddress.substring(accountAddress.length - partLength)}`
);

export function shortNumberFormat(num: number, precision = 2): string {
  const map = [
    { suffix: 'T', threshold: 1e12 },
    { suffix: 'B', threshold: 1e9 },
    { suffix: 'M', threshold: 1e6 },
    { suffix: 'K', threshold: 1e3 },
    { suffix: '', threshold: 1 },
  ];

  const found = map.find(x => Math.abs(num) >= x.threshold);

  return found
    ? (num / found.threshold).toFixed(precision) + found.suffix
    : `${num}`;
}
