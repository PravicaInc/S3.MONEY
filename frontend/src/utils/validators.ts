export const suiAddressRegExp = /^0[xX][a-fA-F0-9]{64}$/;

export const isSuiAddress = (address: string) => suiAddressRegExp.test(address);
