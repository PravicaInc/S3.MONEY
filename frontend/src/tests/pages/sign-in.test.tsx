import { useAutoConnectWallet, useCurrentAccount } from '@mysten/dapp-kit';
import { act, cleanup, RenderResult, screen as testScreen } from '@testing-library/react';

import '@testing-library/jest-dom';

import SignInPage from '@/app/sign-in/page';

import { createMockAccount } from '@/tests/utils/create_mock_wallet_account';
import { renderWithProviders } from '@/tests/utils/render_with_providers';

const renderSignInPageWithProviders = () => renderWithProviders(<SignInPage />);

let renderResult: RenderResult;

describe('Sign-in page:', () => {
  afterAll(cleanup);

  describe('Sign-in page with loading state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'idle');

      renderResult = renderSignInPageWithProviders();
    });

    afterEach(() => renderResult.unmount);

    it('Show loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeVisible();
    });
  });

  describe('Sign-in page with disconnected state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'attempted');

      renderResult = renderSignInPageWithProviders();
    });

    it('Don`t show loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeNull();
    });

    it('Show wallet connect button', () => {
      expect(renderResult.queryByText('Sign In with Sui Wallet')).toBeVisible();
    });

    it('Snapshot', () => {
      expect(renderResult.container).toMatchSnapshot();
    });

    it('Don`t show connect modal', () => {
      const connectButton = renderResult.queryByText('Sign In with Sui Wallet');

      act(() => {
        connectButton?.click();
      });

      expect(testScreen.queryByText('Connect Wallet')).toBeNull();
    });
  });

  describe('Sign-in page with connected state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'attempted');
      jest.mocked(useCurrentAccount).mockImplementation(createMockAccount);

      renderResult = renderSignInPageWithProviders();
    });

    it('Show redirect', () => {
      expect(renderResult.queryByText('Redirecting ...')).toBeVisible();
    });
  });
});
