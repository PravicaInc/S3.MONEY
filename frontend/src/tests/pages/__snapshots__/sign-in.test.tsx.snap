// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-in page: Sign-in page with disconnected state: Snapshot 1`] = `
<div>
  <style>
    [data-dapp-kit] {--dapp-kit-blurs-modalOverlay:blur(0);--dapp-kit-backgroundColors-primaryButton:#F6F7F9;--dapp-kit-backgroundColors-primaryButtonHover:#F0F2F5;--dapp-kit-backgroundColors-outlineButtonHover:#F4F4F5;--dapp-kit-backgroundColors-modalOverlay:rgba(24 36 53 / 20%);--dapp-kit-backgroundColors-modalPrimary:white;--dapp-kit-backgroundColors-modalSecondary:#F7F8F8;--dapp-kit-backgroundColors-iconButton:transparent;--dapp-kit-backgroundColors-iconButtonHover:#F0F1F2;--dapp-kit-backgroundColors-dropdownMenu:#FFFFFF;--dapp-kit-backgroundColors-dropdownMenuSeparator:#F3F6F8;--dapp-kit-backgroundColors-walletItemSelected:white;--dapp-kit-backgroundColors-walletItemHover:#3C424226;--dapp-kit-borderColors-outlineButton:#E4E4E7;--dapp-kit-colors-primaryButton:#373737;--dapp-kit-colors-outlineButton:#373737;--dapp-kit-colors-iconButton:#000000;--dapp-kit-colors-body:#182435;--dapp-kit-colors-bodyMuted:#767A81;--dapp-kit-colors-bodyDanger:#FF794B;--dapp-kit-radii-small:6px;--dapp-kit-radii-medium:8px;--dapp-kit-radii-large:12px;--dapp-kit-radii-xlarge:16px;--dapp-kit-shadows-primaryButton:0px 4px 12px rgba(0, 0, 0, 0.1);--dapp-kit-shadows-walletItemSelected:0px 2px 6px rgba(0, 0, 0, 0.05);--dapp-kit-fontWeights-normal:400;--dapp-kit-fontWeights-medium:500;--dapp-kit-fontWeights-bold:600;--dapp-kit-fontSizes-small:14px;--dapp-kit-fontSizes-medium:16px;--dapp-kit-fontSizes-large:18px;--dapp-kit-fontSizes-xlarge:20px;--dapp-kit-typography-fontFamily:ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";--dapp-kit-typography-fontStyle:normal;--dapp-kit-typography-lineHeight:1.3;--dapp-kit-typography-letterSpacing:1;}
  </style>
  <div
    class="flex flex-col min-h-screen"
  >
    <div
      class="grid md:grid-cols-2 h-full grow"
    >
      <div
        class="bg-white h-full flex items-center justify-center py-5"
      >
        <div
          class="flex flex-col items-center mx-20 max-w-4xl text-center"
        >
          <svg />
          <p
            class="mt-4 text-primary text-[32px] leading-[40px]"
          >
            StableCoin Studio on Sui
          </p>
          <div
            class="mt-5 w-full"
          >
            <button
              class="text-white rounded-xl shadow-button border-actionSecondary border-solid border bg-buttonBgMain relative overflow-hidden after:content-[''] after:bg-buttonBgAfter after:w-full after:h-full after:absolute hover:bg-buttonBgAfter hover:text-actionPrimary transition disabled:bg-slate-300 disabled:bg-none disabled:border-slate-400 disabled:text-white [&>svg>path]:hover:stroke-actionPrimary h-[48px] p-0 w-full text-sm font-semibold flex items-center justify-center"
            >
              Sign In with Sui Wallet
            </button>
          </div>
          <p
            class="mt-5 text-primary lett"
          >
            Donâ€™t have Sui Wallet extension installed?
             
            <a
              class="text-actionPrimary font-semibold"
              href="https://chromewebstore.google.com/detail/sui-wallet/opcgpfmipidbgpenhmajoajpbobppdil"
              rel="noreferrer"
              target="_blank"
            >
              Install Extension
            </a>
          </p>
        </div>
      </div>
      <div
        class="bg-deepPeach h-full border-[10px] border-white flex items-center justify-center py-5"
      >
        <div
          class="text-center mx-5 md:mx-10 lg:mx-20 max-w-4xl"
        >
          <p
            class="text-[40px] leading-[44px] font-semibold text-primary"
          >
            Gateway to Code-Free StableCoin Creation
          </p>
          <p
            class="mt-5 text-tuna text-opacity-70"
          >
            Just connect your Sui wallet and step into the wonderfully simple world of StableCoin Studio on the Sui Blockchain. You'll be able to create and manage your own stable coins with just a few clicks. And the best part? No coding is required! It's a platform designed just for you.
          </p>
          <img
            alt="Sui wallet login screen"
            class="mx-auto mt-14"
            data-nimg="1"
            decoding="async"
            height="525"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fsui_wallet_login_screen.png&w=640&q=75"
            srcset="/_next/image?url=%2Fimages%2Fsui_wallet_login_screen.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fsui_wallet_login_screen.png&w=640&q=75 2x"
            style="color: transparent;"
            width="290"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
